jina/serve/gateway.py

```diff
@@ -172,17 +174,17 @@ async def run_server(self):
         """Run server forever"""
         ...
 
-    @abc.abstractmethod
     async def shutdown(self):
         """Shutdown the server and free other allocated resources, e.g, streamer object, health check service, ..."""
-        ...
+        if self._warmup_task and not self._warmup_task.done():
```

Comment:
try to do this in the `runtime` object and not the gateway, so that it is transparent to the developer implementing the `Gateway`